opensrp-opd/src/test/java/org/smartregister/opd/utils/OpdLookUpUtilsTest.java

```diff
@@ -0,0 +1,119 @@
+                " " + OpdDbConstants.Table.Client.DOB + " , " + OpdDbConstants.Table.Client.BASE_ENTITY_ID + " , " + OpdDbConstants.Table.Client.NATIONAL_ID + " FROM ;";
+        Assert.assertEquals(result, expected_result);
+    }
+
+    @Test
+    public void testGetMainConditionStringWhenEntityMapIsNull() throws Exception {
+        Map<String, String> entityMap = null;
+        expectedException.expect(NullPointerException.class);
+        String result = Whitebox.invokeMethod(OpdLookUpUtils.class, "getMainConditionString", entityMap);
+    }
+
+    @Test
+    public void testGetMainConditionStringWhenEntityMapIsEmpty() throws Exception {
+        Map<String, String> entityMap = new HashMap<>();
+        String result = Whitebox.invokeMethod(OpdLookUpUtils.class, "getMainConditionString", entityMap);
+        Assert.assertEquals(result, "");
+    }
+
+    @Test
+    public void testGetMainConditionStringWhenEntityMapIsWithValue() throws Exception {
+        String firstName = "first_name";
+        String lastName = "last_name";
+        String bht_id = "bht_mid";
+        String national_id = "national_id";
+        Map<String, String> entityMap = new HashMap<>();
+        entityMap.put(firstName, "");
+        entityMap.put(lastName, "");
+        entityMap.put(bht_id, "");
+        entityMap.put(national_id, "");
+        String result = Whitebox.invokeMethod(OpdLookUpUtils.class, "getMainConditionString", entityMap);
+        String actualResult = "national_id Like '%%' AND last_name Like '%%' AND first_name Like '%%' AND bht_mid Like '%%'";
+        Assert.assertNotNull(result, actualResult);
+    }
+
+    @Test
+    public void testClientLookUpWhenContextIsNull() throws Exception {
+        Map<String, String> entityLookUp = new HashMap<>();
+        List<CommonPersonObject> result = Whitebox.invokeMethod(OpdLookUpUtils.class, "clientLookUp", (Object) null, entityLookUp);
+        List<CommonPersonObject> expectedResult = new ArrayList<>();
+        Assert.assertEquals(result, expectedResult);
+    }
+
+    @Test
+    public void testClientLookUpWhenMapIsEmpty() throws Exception {
+        Map<String, String> entityLookUp = new HashMap<>();
+        List<CommonPersonObject> result = Whitebox.invokeMethod(OpdLookUpUtils.class, "clientLookUp", PowerMockito.mock(Context.class), entityLookUp);
+        List<CommonPersonObject> expectedResult = new ArrayList<>();
+        Assert.assertEquals(result, expectedResult);
+    }
+
+    @Test
+    public void testClientLookUpWhenMapIsNotEmptyAndContextIsNotNullWithEmptyTable() throws Exception {
+        OpdLibrary.init(PowerMockito.mock(Context.class), PowerMockito.mock(Repository.class), PowerMockito.mock(OpdConfiguration.class),
+                BuildConfig.VERSION_CODE, 1);
+        PowerMockito.when(OpdUtils.class, "metadata").thenReturn(new OpdMetadata("",
```

Comment:
Does this test actually test a functionality that we would not want to break? Or make sure works a certain way in case of a failure

Reply:
It was testing the expected result when the table if the search/filter will be performed is unavailable